apps/language_server/lib/language_server/providers/workspace_symbols.ex

```diff
@@ -0,0 +1,468 @@
+defmodule ElixirLS.LanguageServer.Providers.WorkspaceSymbols do
```

Comment:
General comment: Searching for "f a" appears to be very slow, probably due to the number of functions (out of 7478 in my index) that have the letter "a" in them. For functions perhaps we need to have a limit of 3 characters in the query before we search. Although we could improve this later.

Reply:
I'm not happy with the performance either. The problem here is that it's not easy to prebuild an index as we don't know the queries upfront. Elixir and erlang data structures are not really helpful for that particular use case. The best option would be to code a custom data structure in a native language but before we do that we need to better know what exactly we need (besides portability issues).
I see two quick solutions.
We could prebuild indexes for all 1 or 2 letter combinations (it should take a few seconds most). The downside is that index update logic will get messier.
We could also preform only exact queries for < 3 letters. This should significantly limit results and serialised data.

Reply:
>  We could also preform only exact queries for < 3 letters. This should significantly limit results and serialised data.

I that sounds like a good approach for now :+1:

Reply:
Done in https://github.com/elixir-lsp/elixir-ls/pull/110/commits/423be36f146e1a7418a679955dca4ef5afd4eb5c. Short query performance is much better now